### PR TITLE
Add catboost to requirements

### LINKED ISSUE #580: Support fancy boosting libraries
There are many, like CatBoost, Light GBM, and XGBoost. As long as they have a scikit learn wrapper for regression we should be able to use them as long as we add it to the requirements-file. Should we add one and test it out?

### PR DESCRIPTION
**Pull request by ryanjdillon**
This PR closes #580, Here I've added `catboost` to the pip requirements, along with an example using the `CatBoostRegressor` in [gordo-test-project #33](https://github.com/equinor/gordo-test-project/pull/33).

### CODE REVIEW

**Review by milesgranger — APPROVED** [review]
I'm good with it, if everyone else is.

**Review by epa095 — APPROVED** [review]
I like it, lets have fun with it!

**Comment by ryanjdillon** [comment]
Sweet :+1:  Needed to update `requirements.txt` after re-rebasing on master, but will merge this in after checks pass again.

**Pull request merged**
